src/compiler/checker.ts

```diff
@@ -4722,6 +4846,1104 @@ namespace ts {
+                    const constructors = serializeSignatures(SignatureKind.Construct, staticType, baseTypes[0], SyntaxKind.Constructor) as ConstructorDeclaration[];
+                    for (const c of constructors) {
+                        // A constructor's return type and type parameters are supposed to be controlled by the enclosing class declaration
+                        // `signatureToSignatureDeclarationHelper` appends them regardless, so for now we delete them here
+                        c.type = undefined;
+                        c.typeParameters = undefined;
+                    }
+                    const indexSignatures = serializeIndexSignatures(classType, baseTypes[0]);
+                    addResult(setTextRange(createClassDeclaration(
+                        /*decorators*/ undefined,
+                        /*modifiers*/ undefined,
+                        localName,
+                        typeParamDecls,
+                        heritageClauses,
+                        [...indexSignatures, ...staticMembers, ...constructors, ...members]
+                    ), symbol.declarations && filter(symbol.declarations, d => isClassDeclaration(d) || isClassExpression(d))[0]), modifierFlags);
+                }
+
+                // Synthesize declarations for a symbol - might be an Interface, a Class, a Namespace, a Type, a Variable (const, let, or var), an Alias
+                // or a merge of some number of those.
+                // An interesting challenge is ensuring that when classes merge with namespaces and interfaces, is keeping
+                // each symbol in only one of the representations
+                // Also, synthesizing a default export of some kind
+                // If it's an alias: emit `export default ref`
+                // If it's a property: emit `export default _default` with a `_default` prop
+                // If it's a class/interface/function: emit a class/interface/function with a `default` modifier
+                // These forms can merge, eg (`export default 12; export default interface A {}`)
+                function serializeSymbolWorker(symbol: Symbol, isPrivate: boolean, propertyAsAlias: boolean) {
+                    const symbolName = unescapeLeadingUnderscores(symbol.escapedName);
+                    const isDefault = symbol.escapedName === InternalSymbolName.Default;
+                    if (isStringANonContextualKeyword(symbolName) && !isDefault) {
+                        // Oh no. We cannot use this symbol's name as it's name... It's likely some jsdoc had an invalid name like `export` or `default` :(
+                        context.encounteredError = true;

```

Comment:
uhhh, how *do* we handle the error with the code as written?

Reply:
Elide the declaration - that's always what we do when `context.encounteredError` is set, as that causes the node returned by the builder to be `undefined`.